### PR TITLE
Add GitHub action to build jars

### PR DESCRIPTION
**Pull request by prantogg**

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set up JDK 17
```
Comment (jiayuasu) [inline .github/workflows/main.yml:34]: Is there a particular reason to use JDK 17? If no, I would prefer JDK 11

**Review by jiayuasu — CHANGES_REQUESTED** [review]

**Comment by jiayuasu** [comment]
Please also give the jar a reasonable name like `wherobots-jdbc` not just `lib-0.1.0`

File: .github/workflows/main.yml (hunk 1 above)
Comment (prantogg) [reply .github/workflows/main.yml:34]: The toolchain in build.gradle specifies the java version as 17. Wanted to be consistent with that. Should I change it to JDK 11 in the build.gradle too?
https://github.com/wherobots/wherobots-jdbc-driver/blob/ee16d5b666f109038b4f0f6a7297120c13b38f86/lib/build.gradle#L38

Comment (jiayuasu) [reply .github/workflows/main.yml:34]: @mpetazzoni can you clarify on this? Is Java 17 really needed?

Comment (jiayuasu) [reply .github/workflows/main.yml:34]: CC @peterfoldes

Comment (mpetazzoni) [reply .github/workflows/main.yml:34]: There shouldn't be a strict need for >= 17, so we could drop down to >= 11.

Note that 11 is on its way out of support though.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        uses: actions/setup-java@v4
+        with:
+          distribution: 'adopt'
+          java-version: '17'
+
+      - name: Verify Gradle installation
+        run: ./gradlew --version
+
+      - name: List initial directory contents
+        run: ls -R
```
Comment (mpetazzoni) [inline .github/workflows/main.yml:44]: Seems unecessary

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+          path: ~/.gradle/caches
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+
+      - name: Make gradlew executable
+        run: chmod +x ./gradlew
+
+      - name: Run build
+        run: ./gradlew build shadowJar
```
Comment (mpetazzoni) [inline .github/workflows/main.yml:58]: You can merge those two:

```yaml
  run: chmod +x gradlew && ./gradlew build shadowJar
```

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+          GRADLE_OPTS: -Dorg.gradle.daemon=false
+          DO_NOT_TRACK: true
+
+      - name: List files in project directory
+        run: ls -R
+        shell: bash
+
+      - name: List files in build directory
+        run: ls -R lib/build
+        shell: bash
```
Comment (mpetazzoni) [inline .github/workflows/main.yml:69]: Superfluous as well

File: .github/workflows/main.yml [hunk 5]
```diff
@@ -0,0 +1,75 @@
+name: Build and Upload JAR
+
+on:
+  push:
+    branches:
+      - master
```
Comment (mpetazzoni) [inline .github/workflows/main.yml:6]: We use `main`

File: .github/workflows/main.yml [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+    branches:
+      - '*'
+    paths:
+      - 'src/**'
+      - 'build.gradle'
+      - '.github/workflows/main.yml'
+
+env:
+  GRADLE_OPTS: -Dorg.gradle.daemon=false
+  DO_NOT_TRACK: true
```
Comment (mpetazzoni) [inline .github/workflows/main.yml:21]: Don't think that's needed here, you pass it to the gradlew build step specifically already

File: lib/build.gradle [hunk 7]
```diff
@@ -44,6 +44,10 @@ java {
 tasks.named('jar') {
     manifest {
         attributes('Implementation-Title': project.name,
-                   'Implementation-Version': project.version)
+                'Implementation-Version': project.version)
```
Comment (mpetazzoni) [inline lib/build.gradle:47]: Revert indent

**Review by mpetazzoni — CHANGES_REQUESTED** [review]

**Review by mpetazzoni — APPROVED** [review]
LGTM as is; we should see about downgrading to JDK 11+ if we can separately.

**Comment by jiayuasu** [comment]
@mpetazzoni The vast majority of the cloud hosting service including Databricks, AWS EMR, Microsoft Fabric are still stuck at Java 8 and some just migrated to Java 11.

**Pull request merged**